openfl/component/aggregator/aggregator.py

```diff
@@ -565,6 +623,49 @@ def send_local_task_results(self, collaborator_name, round_number, task_name,
             task_results.append(tensor_key)
 
         self.collaborator_tasks_results[task_key] = task_results
+
+        # Get all tasks given to the collaborator for current round
+        all_tasks = self.assigner.get_tasks_for_collaborator(
+            collaborator_name, self.round_number
+        )
+        # Check if all given tasks are completed by the collaborator
+        all_tasks_completed = True
+        for task in all_tasks:
+            if hasattr(task, "name"):
+                task = task.name
+            all_tasks_completed = (
+                all_tasks_completed and self._collaborator_task_completed(
+                    collaborator=collaborator_name, task_name=task,
+                    round_num=self.round_number
+                )
+            )
+
+        # If the collaborator has completed ALL tasks for current round,
+        # update collaborators_done
+        if all_tasks_completed:
+            self.collaborators_done.append(collaborator_name)
+            self.logger.info(
+                f"Round: {self.round_number}, Collaborators that have completed all tasks: "
+                f"{self.collaborators_done}"
+            )
+
+        # NOTE: Only for %age based straggler handling policy
+        # If minimum required collaborators have reported results
+        # ignore results from all other collaborators and add them in stragglers list
+        if (
+            not hasattr(self.straggler_handling_policy, "round_start_time") and
+            self.straggler_handling_policy.straggler_cutoff_check(
+                len(self.collaborators_done), self.authorized_cols
+            )
+        ):
```

Comment:
Regardless of the actual policy implementation, shouldn't we always add the lagging collaborators to the `stragglers` list (as soon as the `straggler_cutoff_check` triggers)?

Reply:
The condition is now made generic for all straggler-handling policies.

Reply:
Cool, thanks for fixing this!